lib/active_model/serializer/adapter/json_api/link.rb

```diff
@@ -3,30 +3,45 @@ class Serializer
     module Adapter
       class JsonApi
         class Link
-          def initialize(serializer)
+          def initialize(serializer, value)
             @object = serializer.object
             @scope = serializer.scope
+
+            # Use the return value of the block unless it is nil.
+            if value.respond_to?(:call)
+              self._value = instance_eval(&value)
```

Comment:
Could you please explain why you used an `attr_accessor` for `_value` instead of using a reader and calling `@_value = instance_eval(&value)`? I don't see why `_value` is different than `object` or `scope`.